modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```diff
@@ -0,0 +1,364 @@
+		String url = apiUtilities.getEndPoint();
+		Map<String, String> params = apiUtilities.getRequestParamsMap();
+
+		Map<String, String> responseParams = new HashMap<String, String>();
+		HttpClient httpclient = getCertifiedHttpClient();
+		String payload = buildPayload(params);
+		
+		if (httpMethod.equals(HTTP_METHODS.POST)) {		
+			HttpPost httpPost = new HttpPost(url);
+			HttpPost httpPostWithHeaders = (HttpPost) buildHeaders(httpPost, headers, httpMethod);
+			byte[] postData = payload.getBytes();
+			try {
+				httpPostWithHeaders.setEntity(new ByteArrayEntity(postData));
+				HttpResponse response = httpclient.execute(httpPostWithHeaders);
+				String status = String.valueOf(response.getStatusLine().getStatusCode());
+
+				responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+				responseParams.put(Constants.SERVER_RESPONSE_STATUS, status);
+				return responseParams;
+			} catch (ClientProtocolException e) {
+				throw new IDPTokenManagerException("Invalid client protocol.", e);
+			} catch (IOException e) {
+				responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
+				responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+				                   Constants.INTERNAL_SERVER_ERROR);
+				throw new IDPTokenManagerException("Server connectivity failure.", e);
+			}
+		}
+
+		return responseParams;
+	}
+
+	public static HttpClient getCertifiedHttpClient() throws IDPTokenManagerException {
+		HttpClient client = null;
+		InputStream inStream = null;
+		try {
+			if (Constants.SERVER_PROTOCOL.equalsIgnoreCase("https://")) {
+				KeyStore localTrustStore = KeyStore.getInstance("BKS");
+				inStream =
+						IdentityProxy.getInstance().getContext().getResources()
+						             .openRawResource(R.raw.emm_truststore);
+				localTrustStore.load(inStream, Constants.TRUSTSTORE_PASSWORD.toCharArray());
+
+				SchemeRegistry schemeRegistry = new SchemeRegistry();
+				schemeRegistry.register(new Scheme("http", PlainSocketFactory.getSocketFactory(),
+				                                   80));
+				SSLSocketFactory sslSocketFactory = new SSLSocketFactory(localTrustStore);
+				sslSocketFactory.setHostnameVerifier(SSLSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
+				schemeRegistry.register(new Scheme("https", sslSocketFactory, 443));
+				HttpParams params = new BasicHttpParams();
+				ClientConnectionManager connectionManager =
+						new ThreadSafeClientConnManager(params,
+						                                schemeRegistry);
+
+				client = new DefaultHttpClient(connectionManager, params);
+
+			} else {
+				client = new DefaultHttpClient();
+			}
+
+		} catch (KeyStoreException e) {
+			throw new IDPTokenManagerException("Invalid keystore.", e);
+		} catch (CertificateException e) {
+			throw new IDPTokenManagerException("Invalid certificate.", e);
+		} catch (NoSuchAlgorithmException e) {
+			throw new IDPTokenManagerException("Keystore algorithm does not match.");
```

Comment:
Original exception is swallowed here while initializing and throwing an instance of IDPTokenManagerException.